src/autoload/global.gd

```diff
@@ -52,10 +65,23 @@ var endless_mode := false
 
 var current_music_volume := 0.5
 
+# When the user quits the game, save the game before the engine fully quits
+func _notification(what: int) -> void:
+	if what == NOTIFICATION_WM_CLOSE_REQUEST:
+		save_config()
+		get_tree().quit()
+
+
+func _unhandled_input(event: InputEvent) -> void:
+	if event.is_action_pressed("fullscreen"):
+		# If we're not currently in fullscreen, change to fullscreen. If we are, change back.
+		set_fullscreen(get_window().mode != Window.MODE_FULLSCREEN)
```

Comment:
`get_viewport().set_action_as_handled()` (I think it's get_viewport). Also, I think this should be in `func _input` rather than `func _unhandled_input` to prevent it from conflicting with UI inputs/etc